eppoclient/client.go

```diff
@@ -15,43 +14,61 @@ type EppoClient struct {
 	logger          IAssignmentLogger
 }
 
-func newEppoClient(configRequestor iConfigRequestor, assignmentLogger IAssignmentLogger) *EppoClient {
+func newEppoClient(configRequestor iConfigRequestor, poller *poller, assignmentLogger IAssignmentLogger) *EppoClient {
 	var ec = &EppoClient{}
 
-	var poller = newPoller(10, configRequestor.FetchAndStoreConfigurations)
 	ec.poller = *poller
 	ec.configRequestor = configRequestor
 	ec.logger = assignmentLogger
 
 	return ec
 }
 
-// GetAssignment is maintained for backwards capability. It will return a string value for the assignment.
-func (ec *EppoClient) GetAssignment(subjectKey string, flagKey string, subjectAttributes dictionary) (string, error) {
-	return ec.GetStringAssignment(subjectKey, flagKey, subjectAttributes)
+func (ec *EppoClient) GetBoolAssignment(subjectKey string, flagKey string, subjectAttributes SubjectAttributes, defaultValue bool) (bool, error) {
+	variation, err := ec.getAssignment(subjectKey, flagKey, subjectAttributes, booleanVariation)
+	if variation == nil {
+		return defaultValue, err
+	}
+	result, _ := variation.(bool)
+	return result, err
```

Comment:
I (vaguely) know that this is a common Go pattern, but given that we return the default value when an error occurs, do we need to return the error here? Is the user ever going to care about what `err` is?

Reply:
Mostly for debugging. The error might be helpful to understand why default value is returned